src/main/scala/org/broadinstitute/dsde/firecloud/service/SubmissionService.scala

```diff
@@ -15,20 +14,49 @@ abstract class SubmissionServiceActor extends Actor with SubmissionService {
 
 trait SubmissionService extends HttpService with PerRequestCreator with FireCloudDirectives {
 
-  val routes = postAndGetRoutes
-  lazy val log = LoggerFactory.getLogger(getClass)
-
-  def postAndGetRoutes: Route =
-    pathPrefix("workspaces" / Segment / Segment) {
-      (workspaceNamespace, workspaceName) =>
-        pathPrefixTest("submissions") {
-          val path = FireCloudConfig.Rawls.submissionsUrl.format(workspaceNamespace, workspaceName)
-          passthroughAllPaths("submissions", path)
-        }
-    } ~
+  val routes: Route = {
     path("submissions" / "queueStatus") {
+      get {
+        passthrough(submissionQueueStatusUrl, GET)
+      }
+    } ~
+    pathPrefix("workspaces" / Segment / Segment / "submissions") { (namespace, name) =>
       pathEnd {
-        passthrough(requestCompression = true, FireCloudConfig.Rawls.submissionQueueStatusUrl, HttpMethods.GET)
+        (get | post) {
+          extract(_.request.method) { method =>
+            passthrough(s"$workspacesUrl/$namespace/$name/submissions", method)
+          }
+        }
+      } ~
+      path("validate") {
+        post {
+          extract(_.request.uri.query) { query =>
+            passthrough(Uri(s"$workspacesUrl/$namespace/$name/submissions/validate").withQuery(query), POST)
```

Comment:
this also doesn't accept query params. Maybe you are mixing up query params with post body?

Reply:
Exactly. Thanks for pointing it out!